exercises/bowling/example.rs

```diff
@@ -0,0 +1,136 @@
+
+pub struct BowlingGame {
+    frames: Vec<Frame>,
+}
+
+struct Frame {
+    rolls: Vec<u16>,
+    bonus: Vec<u16>,
+}
+
+impl Frame {
+    fn score(&self) -> u16 {
+        self.roll_score() + self.bonus_score()
+    }
+
+    fn roll_score(&self) -> u16 {
+        self.rolls.iter().sum()
+    }
+
+    fn bonus_score(&self) -> u16 {
+        self.bonus.iter().sum()
+    }
+
+    fn is_valid(&self) -> bool {
+        self.rolls_valid() && self.bonus_valid()
+    }
+
+    fn rolls_valid(&self) -> bool {
+        self.roll_score() <= 10
+    }
+
+    fn bonus_valid(&self) -> bool {
+        if self.bonus_done() {
+            if self.is_spare() {
+                self.bonus_score() <= 10
+            } else if self.is_strike() {
+                if let Some(first) = self.bonus.iter().rev().last() {
+                    if *first == 10 {
+                        self.bonus_score() <= 20
+                    } else {
+                        self.bonus_score() <= 10
+                    }
+                } else {
+                    true
+                }
+            } else {
+                true
```

Comment:
I expect this line (corresponding to "bonus done, but is neither a spare nor a strike") to be unreachable. Correspondingly, I'd prefer `unreachable!` to express that intention. What say you?